src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -3091,6 +3091,244 @@ public void testExecuteDelAndSelect() {
+                                while (rs.next()) {
+                                    String name = rs.getString("NAME");
+                                    assertTrue(name != null, "name should not be null");
+                                    rowCount++;
+                                }
+                                assertEquals(4, rowCount, "Expected 4 rows in result set");
+                            }
+                        }
+                        retval = ps.getMoreResults();
+                    } while (true);
+                }
+            } catch (SQLException e) {
+                fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
+            }
+        }
+
+        /**
+         * Tests PreparedStatement execute for Insert followed by select
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testPreparedStatementExecuteInsertAndSelect() {
+            try (Connection con = getConnection()) {
+                try (PreparedStatement ps = con.prepareStatement("INSERT INTO " + tableName + " (NAME) VALUES(?) SELECT NAME FROM " + tableName + " WHERE ID = 1")) {
+                    ps.setString(1, "test");
+                    boolean retval = ps.execute();
+                    do {
+                        if (!retval) {
+                            int count = ps.getUpdateCount();
+                            if (count == -1) {
+                                // no more results
+                                break;
+                            } else {
+                                assertEquals(count, 1, "update count should have been 1");
+                            }
+                        } else {
+                            // process ResultSet
+                            try (ResultSet rs = ps.getResultSet()) {
+                                if (rs.next()) {
+                                    String val = rs.getString(1);
+                                    assertEquals(val, "test", "read value should have been 'test'");
+                                }
+                            }
+                        }
+                        retval = ps.getMoreResults();
+                    } while (true);
+                }
+            } catch (SQLException e) {
+                fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
+            }
+        }
+
+        /**
+         * Tests PreparedStatement execute for Merge followed by select
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testPreparedStatementExecuteMergeAndSelect() {
+            try (Connection con = getConnection()) {
+                try (PreparedStatement ps = con.prepareStatement("MERGE INTO " + tableName + " AS target USING (VALUES (?)) AS source (name) ON target.name = source.name WHEN NOT MATCHED THEN INSERT (name) VALUES (?); SELECT NAME FROM " + tableName + " WHERE ID = 1")) {
```

Comment:
NIT: break into multiple lines:
String sql = 
    "MERGE INTO " + tableName + " AS target " +
    "USING (VALUES (?)) AS source (name) " +
    "ON target.name = source.name " +
    "WHEN NOT MATCHED THEN INSERT (name) VALUES (?); " +
    "SELECT NAME FROM " + tableName + " WHERE ID = 1";

try (PreparedStatement ps = con.prepareStatement(sql)) {